### ISSUE TITLE
Copper oxidizes faster, Coral dies faster with Pollution, Cold Sweat/TFC provide Atmosphere, Pollution affects atmosphere

### ISSUE THREAD

**Issue by zaafonin** [petrolpark/Destroy#410]
**Description**
Pollution mechanic is the one I'll always cherish whether it's AdPother or Destroy. It's as fun as explosions, and tbh I just like Factorio. I also like how Destroy doesn't deal with individual blocks of pollution; while it doesn't simulate enclosed spaces as accurately, voxel clouds of smoke and sulfur just aren't fun.

**Problem This Addresses**
Pollution should have more impact on the world and the players;

**Acid rain (is it global?)**
- Grass should take a yellow-ish discoloration (savanna/desert-like) because mist and fog are acidic as well afaik; even with no ongoing rainfall grass doesn't exactly like that. Growth-related tick speed (grass, crops) is reduced.
- Because acid rain promotes corrosion, copper should oxidize faster. A more extreme alternative: instead, waxed copper becomes unwaxed. Iron golems take more damage than other entities.
- When it's raining, equipped armor and currently held tools take slight damage; not noticeable but mildly discomfortable.

**Greenhouse (global)**
- [It is well known](https://scholar.google.com/scholar?q=co2+cognitive) CO2 concentration has a negative impact on cognitive performance. Not every greenhouse gas is CO2 but it is still an interesting effect. In Minecraft, this can be simulated by taking away players' experience and/or reducing experience gains from actual collected experience.
- This decline applies to villagers: restocks take longer, levels are gained slower.
- ~~Hard to find but frustrating when noticed: because you're dumb, you get worse enchantments on the table.~~ sounds too specific
- Double-edged sword: in Pollution of the Realms, carbon pollution makes plants grow faster. If pollution is addressed less seriously (like cancer) and more mischievous (like explosives), this feature could make pollution useful (tradeoffs still apply, so it's for player to decide). Also if player is dumber, it's only logical for mob AI to become dumber as well.

**Ozone depletion (global)**
All following suggestions play on the double-edged sword idea:
- Ozone depletion means more UV reaching the surface; more UV means more mutations, so rare mob variations spawn more often, whether it's pink sheep or baby zombies.
- Less need for blacklight when there's more UV available from sunlight.
- Maybe undead mobs start taking damage earlier in the morning?

**Smog (localized)**
In my suggestion, if greenhouse gas inflicts mental fatigue, then smog should inflict physical.
- Applying *Mining Fatigue*, *Weakness* and other status effects might be too serious (I think AdPother does it), but slowing down weapon reloading and item use sounds interesting.
- Saturation (but not hunger!) depletes faster.
- Grass discoloration is too extreme and makes things really ugly because dirt and everything else aren't blackened. IMO AdPother excels at smog by just applying gray sky and fog everywhere.
- Sunlight levels are actually lower, so be careful with mob spawning and prepare for worse plant growth.
I can't really think of a "fun" side effect of smog. Maybe reduced mob vision? Like if they had *Blindness* but not quite.

**Radiation (localized)**
Last time I checked nuclear isn't in the scope of Destroy, but I hope some sub-addon or fairly compatible mod eventually arises. Another option is to integrate with Alex's Caves and its ridiculous nucleepers and nuclear bombs, as right now they're fairly low-consequence. Also the mod has its own Irradiated effect which tbh is pretty balanced (disables regeneration, damages on higher levels).
Because radiation promotes mutations, a lot of ozone depletion (UV) effects can apply, except you can't escape them in the shadow.

If I were to add new pollution types, I'd either split greenhouse gases (different GHGs have their own potency but also different residence time; while water vapor is fairly strong, its residence time is very short; besides, it's quite hard to outvapor the natural concentration) or remove water vapor as a pollutant. Instead, I'd make water vapor its own category that influences frequency of rainfall. I'd also add **heavy metal contamination** that plays on the general idea of farming and reduces edible drops from crops and mobs (wouldn't want to eat leaded meat).

**Generic pollution ideas I couldn't tie to a specific type**
- Natural regeneration is slowed down
- Impaired sleep (can't go sleep right now, try again at midnight or so!)
- Mobs take longer to mature (I'd put this for smog or heavy metal)
- Also longer cooldown between breeding
- Newborn mobs have reduced health (the undead don't :) )

**Minecraft Username**
zaafonin

**Comment by zaafonin** [comment]
Pollution mitigation:
- All types of pollution dissipate and weaken over time
- Vegetation sort of helps with greenhouse gas and smog;
- Smog can be deposited as soot, like vanilla snow; maybe when it's raining? Not a strong idea
- "Acid rain" pollution weakens passively, but when it is raining, this rate is noticeably accelerated.
- Can't do much with ozone depletion and radiation. Persist in the doomed world you have created and wait till it gets better.

Mod compatibility:
I'd love some compat between Destroy and Tough as Nails / Cold Sweat:
- Greenhouse gas leads to higher temperature? A bit obvious and sort of inaccurate (should it happen with a delay? how slowly or rapidly the climate responds? this isn't a climate science mod).
- On the other side, smog leads to lower temperature. Like a nuclear/volcanic winter scenario
Both these "climate-control-via-pollution" changes would make for interesting gameplay, maybe even with space mods if any interesting ones appear.

Talking about temperature... is it possible for bubble cap columns and vats to query for TaN/Cold Sweat temperature?

**Comment by Astrochicken-B** [comment]
Some additions:

This might be an extreme effect, but here in Europe acid rain affected our old buildings and statues, because it dissolves the limestone that they are made of. Perhaps rock-like materials (not granite, andesite and diorite though!) get broken at a slow rate in acid rain. But I think the effect of breaking tree leaves leaving dead forests in its wake is already impressive enough.

CO2 concentration in open air should never be high enough to affect mental capabilities (or everyone will simply get cooked). But particularly hot weather does. When greenhouse gas levels are high, a "hot weather" or "heatwave" debuff might be applied to the player and mobs periodically, which slows down movement, mining and perhaps adds a drunk-slipping like effect. Can be mitigated with a straw hat or a handheld fan. Farm animals don't reproduce and perhaps even take damage if not near a water source (dehydration). Villagers work slower, or don't at all. Also, saplings turn into dead bush occasionally. And maybe non infinite water sources evaporate to punish lazy farmers. Oh yeah, and corals in the ocean die.

I like the idea of heavy metal contamination a lot. I was very sad when the area where I dumped a bunch of my electrolysis mercury was not affected at all. Perhaps harvesting crops / animals in an area with high pollution has an increasing chance to give you 'toxic' or 'contaminated' variants of that food (which poison you, obviously). And in extreme cases, just being there without a hazmat suit does.

**Comment by zaafonin** [comment]
> CO2 concentration in open air should never be high enough to affect mental capabilities (or everyone will simply get cooked). 

Good point. We aren't exactly having 1200ppm on open air no matter what we do. Though it's Minecraft and things are exaggerated; besides, shouldn't indoor CO2 concentrations be higher (and high CO2 effects more frequent) because they start from a higher baseline? Corals dying is a good touch, I really forgot they even existed. High temperature ideas are interesting but technically not pollution-related; though, I'd kill for a TaN-like mod that applied all these nasty effects not just to me.

**Comment by petrolpark** [comment]
Some things I liked or have now been implemented:

- Copper Oxidizing faster will likely be implemented
- Corals dying is also a good idea
- As you may have seen, the "cognitive decline" got added, but caused by lead poisoning rather than carbon dioxide. I liked that idea a lot and @zaafonin you should have been awarded with the suggestion Badge.
- Pollution does naturally decrease now, and there are ways to reduce most types
- Integration with TaN/Cold Sweat is something I may do

Things I don't think would fit terribly well:

- Plants now grow slower with various pollution levels (rather than faster)
- I didn't want to just apply status effects for pollution (I wanted more interesting things, as I think the interconnectedness of Minecraft's world is one of its USPs), so that's unlikely
- I probably won't have smog deposited as soot
- Destruction of blocks, armour and items is a bit extreme, especially with global pollution types

Other things I've not mentioned I am either ambivalent about, or they are very small so low priority.
